### PR TITLE
feat(pod): docker registry auth

### LINKED ISSUE #358: Specify imagePullSecrets for pods
https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/

### PR DESCRIPTION
**Pull request by iliapolo**
Resolves https://github.com/cdk8s-team/cdk8s-plus/issues/358

### CODE REVIEW

File: src/pod.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { Construct } from 'constructs';
 import { ResourceProps, Resource } from './base';
 import { Container, ContainerProps } from './container';
 import * as k8s from './imports/k8s';
+import { DockerConfigSecret } from './secret';
```
Comment (Chriscbr) [inline src/pod.ts:6]: Sweet! Happy to see there's some linking between these components

**Review by Chriscbr — APPROVED** [review]
🚢

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-21/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/582)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-21/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/583)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-20/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/584)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**